Assets/Plugins/Editor/Uplift/DependencyResolution/DependencyNode.cs

```diff
@@ -136,5 +148,199 @@ public int Lowlink
 				lowlink = value;
 			}
 		}
+
+		private void UpdateNodeRequirements(DependencyDefinition newRequirements, string restrictor)
+		{
+			Debug.Log("Updating " + this.name + " requirement " + requirement);
+			Debug.Log(restrictor + " adds new restriction on " + name);
+
+			restrictions[restrictor] = newRequirements.Requirement;
+
+			try
+			{
+				this.requirement = requirement.RestrictTo(restrictions[restrictor]);
+			}
+			catch (IncompatibleRequirementException e)
+			{
+				Debug.Log("Incompatible requirement on node");
+				conflictingRequirementOnNode = newRequirements;
```

Comment:
Here's one thing I haven't understood well. We store those conflicts on the node. We are potentially finding them on several elements on the dependency graph, but seem to only check them on one node. Am I missing something?